restapi/src/main/java/io/stargate/web/docsapi/service/query/ExpressionParser.java

```diff
@@ -157,6 +159,26 @@ public ExpressionParser(ConditionParser predicateProvider) {
     return expressions;
   }
 
+  private void validateFieldConditions(
+      FilterPath filterPath, Collection<BaseCondition> conditions) {
+    // If some conditions imply specific and different value types it is a user error since each
+    // field can have a value of only one type at a time.
+    // Example: `field GT 1` and `field EQ "string"`
+    Set<? extends Class<?>> specificTypes =
```

Comment:
Would it ever be possible to have field condition for multiple fields? 

`field1 GT 1 ... field2 EQ "string"`

Reply:
That example is valid even now. Current change applies only to collections of conditions on one particular field.